src/runner/fixture-hook-controller.ts

```diff
@@ -59,30 +59,52 @@ export default class FixtureHookController {
         return !!item && item.runningFixtureBeforeHook;
     }
 
+    private async _runFixtureBeforeHook (item: FixtureState, fn: Function): Promise<boolean> {
+        const shouldRunBeforeHook = !item.started && fn;
+
+        if (!shouldRunBeforeHook)
+            return !item.fixtureBeforeHookErr;
+
+        item.runningFixtureBeforeHook = true;
+
+        try {
+            await fn(item.fixtureCtx);
+        }
+        catch (err) {
+            item.fixtureBeforeHookErr = processTestFnError(err);
+        }
+
+        item.runningFixtureBeforeHook = false;
+
+        return !item.fixtureBeforeHookErr;
+    }
+
+    private async _runFixtureAfterHook (item: FixtureState, fn: Function | null, testRun: TestRun): Promise<void> {
+        if (!fn)
+            return;
+
+        testRun.phase = TEST_RUN_PHASE.inFixtureAfterHook;
+
+        try {
+            await fn(item.fixtureCtx);
+        }
+        catch (err) {
+            testRun.addError(processTestFnError(err));
+        }
+    }
+
     public async runFixtureBeforeHookIfNecessary (testRun: TestRun): Promise<boolean> {
         const fixture = testRun.test.fixture;
         const item    = this._getFixtureMapItem(testRun.test);
 
         if (item) {
-            const shouldRunBeforeHook = !item.started && fixture.beforeFn;
+            const success = await this._runFixtureBeforeHook(item, fixture.globalBeforeFn as Function)
+                            && await this._runFixtureBeforeHook(item, fixture.beforeFn as Function);
 
             item.started = true;
```

Comment:
in the previous version the `started` property was set before the hook execution